pkg/api/v1/router_server_secrets.go

```diff
@@ -0,0 +1,149 @@
+package serverservice
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+
+	"go.hollow.sh/serverservice/internal/dbtools"
+	"go.hollow.sh/serverservice/internal/models"
+)
+
+func (r *Router) serverSecretGet(c *gin.Context) {
+	mods := []qm.QueryMod{
+		models.ServerSecretWhere.ServerID.EQ(c.Param("uuid")),
+		qm.InnerJoin(fmt.Sprintf("%s as t on t.%s = %s.%s",
+			models.TableNames.ServerSecretTypes,
+			models.ServerSecretTypeColumns.ID,
+			models.TableNames.ServerSecrets,
+			models.ServerSecretColumns.ServerSecretTypeID,
+		)),
+		qm.Where(fmt.Sprintf("t.%s=?", models.ServerSecretTypeColumns.Slug), c.Param("slug")),
+		qm.Load(models.ServerSecretRels.ServerSecretType),
+	}
+
+	dbS, err := models.ServerSecrets(mods...).One(c.Request.Context(), r.DB)
+	if err != nil {
+		dbErrorResponse(c, err)
```

Comment:
What kind of errors could we get here? I'm thinking about sanitizing the error message so we don't leak secrets in the logs.

Reply:
The errors here are usually either a connection to the database failed or something went wrong talking to it or the record doesn't exist. The dbErrorResponse handles if it is missing and returning a 404 instead of an error

Reply:
great!